File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/network/RequestResultListener.java
package com.vladislavmyasnikov.courseproject.data.network;

import retrofit2.Call;
import retrofit2.Response;

public interface RequestResultListener<T> {

    void onFailure(Call<T> call, Throwable e);
    void onResponse(Call<T> call, Response<T> response);
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/network/RequestResultListener.kt
package com.vladislavmyasnikov.courseproject.data.network

import retrofit2.Call
import retrofit2.Response

interface RequestResultListener<T> {

    fun onFailure(call: Call<T>, e: Throwable)
    fun onResponse(call: Call<T>, response: Response<T>)
}
